src/main/osd/osd_elements.c

```diff
@@ -1309,26 +1279,26 @@ static void osdElementWarnings(osdElementParms_t *element)
             i++;
         }
 
-        escWarningMsg[pos] = '\0';
+        escWarningMsg[pos++] = '\0';
 
         if (escWarningCount > 0) {
-            osdFormatMessage(element->buff, OSD_FORMAT_MESSAGE_BUFFER_SIZE, escWarningMsg);
+            memcpy(element->buff, escWarningMsg, pos);
```

Comment:
Could probably use `tfp_sprintf(element->buff, "%s", escWarningMsg) here to keep it consistent.

Reply:
I was in doubt of how to do it and finally I opted by the memcpy. But I will change it, as you say is more consistent.

Reply:
It may be best to use buff directly, avoiding `escWarningMsg` above. Then simply return only if warning was produced.